packages/puppeteer-core/src/common/ElementHandle.ts

```diff
@@ -412,6 +421,61 @@ export class ElementHandle<
     return this.waitForSelector(`xpath/${xpath}`, options);
   }
 
+  /**
+   * Converts the current handle to the given element type.
+   *
+   * @example
+   *
+   * ```ts
+   * const element: ElementHandle<Element> = await page.$(
+   *   '.class-name-of-anchor'
+   * );
+   * // DO NOT DISPOSE `element`, this will be always be the same handle.
+   * const anchor: ElementHandle<HTMLAnchorElement> = await element.toElement(
+   *   'a'
+   * );
+   * ```
+   *
+   * @example
+   * Garbage handling is automatically handled on error, but you can turn this off.
+   *
+   * ```ts
+   * const element: ElementHandle<Element> = await page.$(
+   *   '.class-name-of-anchor'
+   * );
+   * try {
+   *   const anchor = await element.toElement('svg', {dispose: false});
+   * } catch (err) {
+   *   // You need to handle the trash since you specified `dispose: false`
+   *   element.dispose();
+   *   throw new err();
+   * }
+   * ```
+   *
+   * @param tagName - The tag name of the desired element type.
+   * @param options - Configures the behavior of this method.
+   * @throws An error if the handle does not match. **This will dispose the
+   * current handle.** You can disable this behavior in the options.
+   */
+  async toElement<
```

Comment:
Is there a bug/issue requesting this? where would we use it? can it be internal?

Reply:
This is something I wanted internally after trying to upstream a new puppeteer version. There are a lot of errors that can be fixed by `as ...`, but this is prone to misuse and not descriptive if something does break. This is also a solution to:
 - https://github.com/puppeteer/puppeteer/issues/8579
 - https://github.com/puppeteer/puppeteer/issues/9280